drivers/serial/uart_nrfx_uarte.c

```diff
@@ -0,0 +1,606 @@
+
+#if (CONFIG_UART_NRFX_TX_BUFFER_SIZE > UART_MAX_TX_BUFF_SIZE)
+	#define UART_FIFO_TX_BUFFER_SIZE	UART_MAX_TX_BUFF_SIZE
+#else
+	#define UART_FIFO_TX_BUFFER_SIZE	CONFIG_UART_NRFX_TX_BUFFER_SIZE
+#endif /* (CONFIG_UART_NRFX_TX_BUFFER_SIZE > UART_MAX_TX_BUFF_SIZE) */
+
+/* Device data structure */
+struct uarte_nrfx_data {
+	u8_t rx_data;
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_callback_t cb;	/**< Callback function pointer */
+	u8_t tx_buffer[UART_FIFO_TX_BUFFER_SIZE];
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+};
+
+/**
+ * @brief Structure for UARTE configuration.
+ */
+struct uarte_nrfx_config {
+	NRF_UARTE_Type *uarte_regs; /* Instance address */
+};
+
+struct uarte_init_config {
+	u32_t  pseltxd; /* TXD pin number */
+	u32_t  pselrxd; /* RXD pin number. */
+	u32_t  pselcts; /* CTS pin number. */
+	u32_t  pselrts; /* RTS pin number. */
+
+	nrf_uarte_hwfc_t      hwfc;     /* Flow control configuration. */
+	nrf_uarte_parity_t    parity;   /* Parity configuration. */
+	nrf_uarte_baudrate_t  baudrate; /* Baudrate. */
+};
+
+
+static inline struct uarte_nrfx_data *get_dev_data(struct device *dev)
+{
+	return dev->driver_data;
+}
+
+static inline const struct uarte_nrfx_config *get_dev_config(struct device *dev)
+{
+	return dev->config->config_info;
+}
+
+static inline NRF_UARTE_Type *get_uarte_instance(struct device *dev)
+{
+	const struct uarte_nrfx_config *config = get_dev_config(dev);
+
+	return config->uarte_regs;
+}
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+/**
+ * @brief Interrupt service routine.
+ *
+ * This simply calls the callback function, if one exists.
+ *
+ * @param arg Argument to ISR.
+ *
+ * @return N/A
+ */
+static void uarte_nrfx_isr(void *arg)
+{
+	struct device *dev = arg;
+	const struct uarte_nrfx_data *data = get_dev_data(dev);
+
+	if (data->cb) {
+		data->cb(dev);
+	}
+}
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+
+/**
+ * @brief Set the baud rate
+ *
+ * This routine set the given baud rate for the UARTE.
+ *
+ * @param dev UARTE device struct
+ * @param baudrate Baud rate
+ *
+ * @return 0 on success or error code
+ */
+static int baudrate_set(struct device *dev, u32_t baudrate)
+{
+	nrf_uarte_baudrate_t nrf_baudrate; /* calculated baudrate divisor */
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	switch (baudrate) {
+	case 300:
+		/* value not supported by Nordic HAL */
+		nrf_baudrate = 0x00014000;
+		break;
+	case 600:
+		/* value not supported by Nordic HAL */
+		nrf_baudrate = 0x00027000;
+		break;
+	case 1200:
+		nrf_baudrate = NRF_UARTE_BAUDRATE_1200;
+		break;
+	case 2400:
+		nrf_baudrate = NRF_UARTE_BAUDRATE_2400;
+		break;
+	case 4800:
+		nrf_baudrate = NRF_UARTE_BAUDRATE_4800;
+		break;
+	case 9600:
+		nrf_baudrate = NRF_UARTE_BAUDRATE_9600;
+		break;
+	case 14400:
+		nrf_baudrate = NRF_UARTE_BAUDRATE_14400;
+		break;
+	case 19200:
+		nrf_baudrate = NRF_UARTE_BAUDRATE_19200;
+		break;
+	case 28800:
+		nrf_baudrate = NRF_UARTE_BAUDRATE_28800;
+		break;
+	case 38400:
```

Comment:
@jarz-nordic  The previous driver supported speeds 31250 and 56000. Are we deliberately dropping support for these speeds?

Reply:
@jarz-nordic I have double-checked that. The driver _nrf5 for Zephyr did not support the speeds mentioned. However, the NRFX/HAL do support and the SDK also provides support for these speeds. The following is a snippet of the SDK configuration file:

// <o> NRFX_UARTE_DEFAULT_CONFIG_BAUDRATE  - Default Baudrate
// <323584=> 1200 baud 
[...]
// <7716864=> 28800 baud 
// <8388608=> 31250 baud   <-----
// <10289152=> 38400 baud 
// <15007744=> 56000 baud   <-----
// <15400960=> 57600 baud 
[...]

Reply:
fixed